### PR TITLE
FOC-PW-3214 Add bacs payment method

### PR DESCRIPTION
**Pull request by zurmalik**
**Description**
Add BACS payment method.

**Tested scenarios**
A successful payment was created which resulted in a response with a correct redirect url. This URL could be used to further enter the info to complete the payment.

**Fixed issue**:  PW-3214

### CODE REVIEW

File: src/test/java/com/adyen/CheckoutTest.java [hunk 1]
```diff
@@ -798,6 +798,37 @@ public void TestBilldeskOnlineDetails(){
...
+    @Test
+    public void TestBacsDirectDebitDetails() {
+        BacsDirectDebitDetails bacsDirectDebitDetails = new BacsDirectDebitDetails();
+
+        PaymentsRequest paymentsRequest = createBacsDirectDebitDetails();
+        paymentsRequest.setPaymentMethod(bacsDirectDebitDetails);
+
+        String jsonRequest = PRETTY_PRINT_GSON.toJson(paymentsRequest);
+
+        jsonRequest += "";
```
Comment (rkewlani) [inline src/test/java/com/adyen/CheckoutTest.java:810]: why this is needed?

File: src/test/java/com/adyen/CheckoutTest.java [hunk 2]
```diff
@@ -798,6 +798,37 @@ public void TestBilldeskOnlineDetails(){
...
+
+        assertEquals("{\n"
+                + "  \"amount\": {\n"
+                + "    \"value\": 1000,\n"
+                + "    \"currency\": \"GBP\"\n"
+                + "  },\n"
+                + "  \"merchantAccount\": \"MagentoMerchantTest2\",\n"
```
Comment (Aleffio) [inline src/test/java/com/adyen/CheckoutTest.java:817]:
```suggestion
                + "  \"merchantAccount\": \"TestMerchant\",\n"
```

File: src/test/java/com/adyen/CheckoutTest.java [hunk 3]
```diff
@@ -1649,6 +1680,21 @@ protected PaymentsRequest createPaymentsCheckoutRequest() {
...
+     * Returns a sample PaymentsRequest object with test data
+     */
+    protected PaymentsRequest createBacsDirectDebitDetails() {
+        PaymentsRequest paymentsRequest = new PaymentsRequest();
+
+        paymentsRequest.setReference("Your order number");
+        paymentsRequest.setAmount(createAmountObject("GBP", 1000L));
+
+        paymentsRequest.setReturnUrl("https://your-company.com/...");
+        paymentsRequest.setMerchantAccount("MagentoMerchantTest2");
```
Comment (Aleffio) [inline src/test/java/com/adyen/CheckoutTest.java:1693]:
```suggestion
        paymentsRequest.setMerchantAccount("TestMerchant");
```

File: src/test/java/com/adyen/CheckoutTest.java (hunk 1 above)
Comment (zurmalik) [reply src/test/java/com/adyen/CheckoutTest.java:810]: This is in fact not needed. It is being removed in the next commit. 
I hope the next commit can be made part of this same pull request ?

**Review by Aleffio — APPROVED** [review]

**Review by andrei-dediu — APPROVED** [review]

**Pull request merged**